File: Preparation Kit/Week 1/3_Time_Conversion.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'timeConversion' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string timeConversion(string s) {
    string hour = s.substr(0, 2);
    string minute = s.substr(3, 2);
    string second = s.substr(6, 2);
    string period = s.substr(8, 2);

    int hh = stoi(hour);

    if (period == "AM") {
        if (hh == 12) {
            hh = 0;
        }
    } else {
        if (hh != 12) {
            hh += 12;
        }
    }

    stringstream result;
    result << setfill('0') << setw(2) << hh << ":"
           << minute << ":"
           << second;

    return result.str();
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = timeConversion(s);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: Preparation Kit/Week 1/3_Time_Conversion.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'timeConversion' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def timeConversion(s):
    # Write your code here
    period = s[-2:]
    hh, mm, ss = s[:-2].split(':')
    
    if period == 'AM':
        if hh == '12':
            hh = '00'
    else:
        if hh != '12':
            hh = str(int(hh) + 12)

    return f"{hh}:{mm}:{ss}"

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    fptr.write(result + '\n')

    fptr.close()
